src/gluonts/model/rotbaum/_preprocess.py

```diff
@@ -306,6 +341,41 @@ def _pre_transform(cls, time_series_window) -> Tuple:
             },
         )
 
+    def encode_one_hot(self, feat: int, cardinality: int) -> List[int]:
+        result = [0] * cardinality
+        result[feat] = 1
+        return result
+
+    def encode_one_hot_all(self, feat_list: List):
+        # asserts that the categorical features are label encoded
+        np_feat_list = np.array(feat_list)
+        assert all(np.floor(np_feat_list) == np_feat_list)
+        del np_feat_list
+
+        encoded = starmap(
+            self.encode_one_hot, zip(feat_list, self.cardinality)
+        )
+        encoded_chain = chain.from_iterable(encoded)
+        return list(encoded_chain)
+
+    def create_cardinalities(self, time_series):
+        if "feat_static_cat" not in time_series[0]:
+            return []
+        mat = [None] * len(time_series)
+        for i, elem in enumerate(time_series):
+            mat[i] = np.array(elem["feat_static_cat"])
+        mat = np.array(mat)
+        assert (mat[i].size == mat[0].size for i in range(len(mat)))
```

Comment:
There should be no need to check for this, since proper arrays have this as a requirement.